app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,40 @@
+class CommentsController < ApplicationController
+  before_action :authenticate_user!
```

Comment:
You have added this line in most of the controller. There is a better way of doing so.

Reply:
Add this line in application controller.